src/js/components/Select/Select.js

```diff
@@ -262,22 +228,16 @@ const Select = forwardRef(
       return undefined;
     }, [value, valueLabel]);
 
-    // if labelKey is a function and valueLabel is not defined
-    // we should use the labelKey function to display the
-    // selected value
-    const displayLabelKey = useMemo(() => {
-      const optionLabelKey = applyKey(
-        allOptions[optionIndexesInValue[0]],
-        labelKey,
-      );
-      if (
-        !selectValue &&
-        optionIndexesInValue.length === 1 &&
-        typeof optionLabelKey === 'object'
-      )
-        return optionLabelKey;
-      return undefined;
-    }, [labelKey, allOptions, optionIndexesInValue, selectValue]);
+    const displayLabelKey = useMemo(
```

Comment:
As an example of custom hooks, this could be `useDisplayLabelKey(labelKey, allOptions, optionIndexesInValue, selectValue)`. Not required, just illustrating.

Reply:
Good to know, thanks for the example